processor/src/trace/tests/chiplets/hasher.rs

```diff
@@ -637,11 +654,22 @@ fn fill_state_from_decoder(trace: &ExecutionTrace, state: &mut HasherState, row:
     }
 }
 
-/// Absorbs the elements specified in the decoder's execution trace helper columns at the specified
-/// row into the provided HasherState.
-fn absorb_state_from_decoder(trace: &ExecutionTrace, state: &mut HasherState, row: usize) {
-    for (i, col_idx) in DECODER_HASHER_STATE_RANGE.enumerate() {
-        state[CAPACITY_LEN + i] = trace.main_trace.get_column(col_idx)[row];
+/// Extract the control block domain from the execution trace.  This is achieved using
+/// bit composition of the op bits + 1. [bit_0 * 2**0 + bit_1 * 2**1 + ... + bit_6 * 2**6 + 1]
+fn extract_control_block_domain_from_trace(trace: &ExecutionTrace, row: usize) -> Felt {
+    let mut composed_op_bits = Felt::new(1);
```

Comment:
Are you sure this should be initialised with 1 instead of 0?

Reply:
It was suggested that the opcode value should be computed as `[bit_0 * 2**0 + bit_1 * 2**1 + ... + bit_6 * 2**6 + 1]`.  This is to avoid having a 0 value opcode.  I'm not sure this is necessary and is inconsistent with [other resources](https://0xpolygonmiden.github.io/miden-vm/design/stack/op_constraints.html#high-degree-operations) which do not include the addition.  I'll have this clarified.  Good catch.

Reply:
@frisitano The reason we added 1 in the case of the [operation labels](https://0xpolygonmiden.github.io/miden-vm/design/chiplets/main.html#operation-labels) was because it was problematic for the multiset check to have one of the operation labels as `0`. I don't think we have that issue here, so we shouldn't need to add 1 in this case.

Reply:
Thank you for clarifying @grjte, that makes sense.